### PR TITLE
Parallel tests are consistently used in integration tests

### LINKED ISSUE #2394: Fix handling of internal errors
#2393, #2410, #2564.

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #2394.
* tests are run parallel on top level test when they can.
* sub tests are run parallel consistently.
* fix incorrect compat test runner.

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [ ] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [x] I ensured that PR title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/explain_compat_test.go [hunk 1]
```diff
@@ -67,28 +67,28 @@ func testExplainCompatError(t *testing.T, testCases map[string]explainCompatTest
 			t.Run(targetCollection.Name(), func(t *testing.T) {
 				t.Helper()
 
-				explainParams := bson.D{
-					{tc.command, targetCollection.Name()},
-				}
+				explainTarget := bson.D{{tc.command, targetCollection.Name()}}
+				explainCompat := bson.D{{tc.command, compatCollection.Name()}}
```
Comment (chilagrow) [inline integration/explain_compat_test.go:71]: I noticed that both target and compat were using `targetCollection`. compat should use compatCollection, target should use targetCollection.

**Review by rumyantseva — DISMISSED** [review]
Nice! As discussed in the call, this should do the trick!

File: integration/commands_administration_test.go [hunk 2]
```diff
@@ -769,6 +769,8 @@ func TestCommandsAdministrationDBStatsEmptyWithScale(t *testing.T) {
 func TestCommandsAdministrationServerStatus(t *testing.T) {
 	setup.SkipForTigris(t)
 
+	t.Parallel()
```
[the remark this answers is not in the document]
Comment (w84thesun) [reply integration/commands_administration_test.go]: Fixed.

**Review by w84thesun — APPROVED** [review]

File: integration/commands_administration_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (chilagrow) [reply integration/commands_administration_test.go]: Thank you both :hugs:

**Pull request merged**